controller/eks-cluster-config-handler.go

```diff
@@ -353,11 +353,16 @@ func validateUpdate(config *eksv1.EKSClusterConfig) error {
 		if clusterVersion.EQ(*version) {
 			continue
 		}
-		if clusterVersion.Minor-version.Minor == 1 {
+		if (clusterVersion.Minor < 25 && clusterVersion.Minor-version.Minor <= 2) || (clusterVersion.Minor >= 25 && clusterVersion.Minor-version.Minor <= 3) {
 			continue
 		}
-		errs = append(errs, fmt.Sprintf("versions for cluster [%s] and nodegroup [%s] not compatible: all nodegroup kubernetes versions"+
-			"must be equal to or one minor version lower than the cluster kubernetes version", aws.ToString(config.Spec.KubernetesVersion), aws.ToString(ng.Version)))
+		if clusterVersion.Minor < 25 {
+			errs = append(errs, fmt.Sprintf("versions for cluster [%s] and node group [%s] are not compatible: for clusters running Kubernetes < 1.25 the "+
+				"node group version may only be up to two minor versions older than the cluster version", aws.ToString(config.Spec.KubernetesVersion), aws.ToString(ng.Version)))
```

Comment:
When this policy was changed, official documentation is still saying that we can't skip any minor version during upgrade? https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade/

Reply:
The control plane versions have to be upgraded without skipping minor versions - but this check is comparing the control plane version against the node's kubelet version.